### DISCUSSION TITLE
API issue - Swagger Docs Page is empty

### DISCUSSION THREAD

**Discussion by danieldrinhausen** [kimai/kimai#4819] [On-Premise: Installation & Updates]
Hey there,

I have installed Kimai on my hosted environment, everything is working well except the API.

When navigating to my user profile -> API access and clicking there on the swagger docs it takes me to an empty page. If I go to the direct api page (https://myurl/api) it gives me a {"code":404,"message":"Not Found"}.

![CleanShot 2024-04-29 at 11 21 14@2x](https://github.com/kimai/kimai/assets/152017997/123cc8a2-f9c0-4bb7-b153-511159bb1835)

Any idea what could be wrong and how I can get it fixed? Thanks!

**Comment by kevinpapst** [comment]
>  takes me to an empty page

Which URL? Whats in the log (System > Doctor and Webserver)?

**Reply by danieldrinhausen** [reply]
Url is: https://myurl/api/doc

Doctor has no entry for today. Webserver I need to check where I can find that information.

**Reply by kevinpapst** [reply]
Are you using Apache or nginx? 
There might to be some kind of bug that cannot be identified by now:
https://github.com/kimai/kimai/discussions/4797#discussioncomment-9258687

**Reply by danieldrinhausen** [reply]
Apache and also running 2.15 version. Maybe it's related.

**Comment by danieldrinhausen** [comment]
Hey, I have seen there is a new version 2.16, thanks for that. Unfortunately that hasn't solved my issue, do you have any idea how I get tackle that?

**Reply by kevinpapst** [reply]
You need to share more infos about your environment.
The API screen needs to be able to write cache files when it is opened. A white screen usually means some very nasty 500 error.
This one would definitely be logged. 
Are you sure your webserver can write logfiles and cache files?

**Reply by danieldrinhausen** [reply]
Thanks Kevin, much appreciated. I don't have access to the webserver logs from what I can see but it's probably related to that. When navigating to http://mydomainkimait/api I get an {"code":404,"message":"Not Found"}. 

Also tried to install a fresh Kimai somewhere else in that environment with the exact same result. All is working except the API.

Have to get in touch with the provider to figure out what I can do and what I can't.

**Reply by danieldrinhausen** [reply]
Also the error logs from the web server doesn't show anything related (which is strange and why I have opened a ticket).

Could it be related to file permissions by any chance? As im getting the 404 error.

For the following part I can't execute the command for www-data . - Could that be a reason? How can I figure out the right name of that folder for my environment?

chown -R :www-data .
chmod -R g+r .
chmod -R g+rw var/

// Probably unrelated as everything looks good from the kimai doctors perspective
![CleanShot 2024-05-03 at 09 07 56](https://github.com/kimai/kimai/assets/152017997/0f277992-f340-45f7-9fe5-ddcb0f75dbf7)

**Reply by danieldrinhausen** [reply]
Got feedback from the provider.

> bzgl. https://kimaidomain/api gibt es keinerlei Logeintrag aufgrund des 404.
> 
> Hier sollten Sie sich einmal Ihre Umleitungen genauer ansehen.
> 
> In den Logfiles kann ich legiglich ein "Abort request processing " für Ihre IP finden.
> 
> 2024-05-03 07:51:22.494816 [INFO] [30903] [62.228.51.182:65393:HTTP2-1#APVH_kimaidomain:443] Abort request processing by PID:8008, kill: 1, begin time: 0, sent time: 0, req processed: 0
> 2024-05-03 08:54:32.266512 [INFO] [30903] [62.228.51.182:49349:HTTP2-9#APVH_kimaidomain:443] Abort request processing by PID:1910, kill: 1, begin time: 1, sent time: 1, req processed: 0

Any other idea left? Thanks in advance!

**Comment by danieldrinhausen** [comment]
With the latest update this issue seems to be resolved. Thanks.
